src/group_theory/perm/basic.lean

```diff
@@ -287,6 +287,20 @@ equiv.ext $ λ ⟨x, hx⟩, by { dsimp [subtype_perm, of_subtype],
 
 @[simp] lemma default_perm {n : Type*} : default (equiv.perm n) = 1 := rfl
 
+/-- Permutations on a subtype are equivalent to permutations on the original type that fix pointwise
+the rest. -/
+protected def perm.subtype_equiv (p : α → Prop) [decidable_pred p] :
+  perm (subtype p) ≃ {f : perm α // ∀ a, ¬p a → f a = a} :=
+{ to_fun := λ f, ⟨f.of_subtype, λ a, f.of_subtype_apply_of_not_mem⟩,
+  inv_fun := λ ⟨f, hf⟩, (f : perm α).subtype_perm
```

Comment:
Pattern matching lambdas don't reduce well. Can you use `f` and `f.prop` instead of `f` and `hf`?